### PR TITLE
RDBC-698 Python client - timeseries round 2

### PR DESCRIPTION
**Pull request by poissoncorp**
https://issues.hibernatingrhinos.com/issue/RDBC-698/Python-client-timeseries-round-2

### CODE REVIEW

File: ravendb/documents/operations/time_series.py [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
         self.retention_time = retention_time
         self.aggregation_time = aggregation_time
 
+        self.name = name
+
     @classmethod
     def from_json(cls, json_dict: Dict[str, Any]) -> TimeSeriesPolicy:
         return cls(
-            json_dict["Name"],
+            json_dict["Name"],  # todo: Invalid deserialization
```
Comment (ml054) [inline ravendb/documents/operations/time_series.py:49]: ?

File: ravendb/documents/operations/time_series.py [hunk 2]
```diff
@@ -55,7 +56,7 @@ def get_time_series_name(self, raw_name: str) -> str:
 
     def to_json(self) -> Dict[str, Any]:
         return {
-            "Name": self.name,
+            "Name": self.name,  # todo: Invalid serialization
```
Comment (ml054) [inline ravendb/documents/operations/time_series.py:59]: ?

File: ravendb/documents/session/document_session.py [hunk 3]
```diff
@@ -168,7 +171,7 @@ def execute_all_pending_lazy_operations(self) -> ResponseTimeInformation:
         requests = []
         for i in range(len(self._pending_lazy_operations)):
             # todo: pending lazy operation create request - WIP
-            req = self._pending_lazy_operations[i].create_request()
+            req = self._pending_lazy_operations[i].create_arequest()
```
Comment (ml054) [inline ravendb/documents/session/document_session.py:174]: typo?

File: ravendb/documents/session/time_series.py [hunk 4]
```diff
@@ -186,60 +222,116 @@ def from_json(cls, json_dict: Dict[str, Any]):
...
         )
 
 
+class ITimeSeriesValuesBindable(abc.ABC):
+    @abc.abstractmethod
+    def get_time_series_mapping(self) -> Dict[int, Tuple[str, Optional[str]]]:
+        # return a dictionary {index of time series value - (name of 'float' field, label)}
+        # e.g. return {0 : ('heart', 'Heartrate'), 1: ('bp', 'Blood Pressure')}
+        # for some class that has 'heart' and 'bp' float fields
+        raise NotImplementedError()
```
Comment (ml054) [inline ravendb/documents/session/time_series.py:241]: ?

File: ravendb/tests/jvm_migrated_tests/issues_tests/test_ravenDB_14994.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            session.save_changes()
+
+        get = self.store.operations.send(GetTimeSeriesOperation(document_id, "HeartRate"))
+        self.assertIsNone(get)
+
+        with self.store.open_session() as session:
+            self.assertIsNone(session.time_series_for(document_id, "HeartRate").get())
+
+    def test_get_on_empty_range_should_return_empty_array(self):
+        document_id = "users/gracjan"
```
Comment (ml054) [inline ravendb/tests/jvm_migrated_tests/issues_tests/test_ravenDB_14994.py:25]: it was  "users/ayende" in original source code :)

File: ravendb/documents/session/time_series.py (hunk 4 above)
Comment (poissoncorp) [reply ravendb/documents/session/time_series.py:241]: user has to implement it https://issues.hibernatingrhinos.com/issue/RDBC-698/Python-client-timeseries-round-2#focus=Comments-67-1046450.0-0

File: ravendb/documents/operations/time_series.py (hunk 1 above)
Comment (poissoncorp) [reply ravendb/documents/operations/time_series.py:49]: fixed in the upcoming commits in this pr https://issues.hibernatingrhinos.com/issue/RDBC-766/Python-client-time-series-3

File: ravendb/documents/operations/time_series.py (hunk 2 above)
Comment (poissoncorp) [reply ravendb/documents/operations/time_series.py:59]: fixed in the upcoming commits in this pr https://issues.hibernatingrhinos.com/issue/RDBC-766/Python-client-time-series-3

File: ravendb/documents/session/document_session.py (hunk 3 above)
Comment (poissoncorp) [reply ravendb/documents/session/document_session.py:174]: typo, fixed in the upcoming commits in this pr https://issues.hibernatingrhinos.com/issue/RDBC-766/Python-client-time-series-3

**Pull request merged**